### PR TITLE
Merge substra-ui back into substra-frontend

### PR DESCRIPTION
**Pull request by maeldebon**
This PR aims to merge substra-ui back into substra-frontend.
It is related to [this issue](https://github.com/SubstraFoundation/substra-frontend/issues/63).

Remaining tasks:
- [x] Move the icons
- [x] Move globalStyle
- [x] Move the components
- [x] Move propTypes
- [x] Move and merge variables
- [x] Update all references 
- [x] Clean the code
- [x] Move the .storybook.js files
- [x] Move Storybook and Jest configurations
- [x] Make everything running

### CODE REVIEW

File: src/app/business/common/components/searchBar/searchBar.js [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+        box-shadow: 0 0 3pt 3pt ${darkSkyBlue};
+    }
+`;
+
+// use getRootProps https://github.com/paypal/downshift#getrootprops
+const SearchInputWrapper = ({innerRef, ...rest}) => (
+    <div
+        className={searchInputWrapper}
+        ref={innerRef}
+        key={this}
```
Comment (jmorel) [inline src/app/business/common/components/searchBar/searchBar.js:51]: This looks like an addition from the code in substra-ui, could you explain why it is needed?

File: src/app/business/common/components/searchBar/searchBar.js [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+            <SearchInputWrapper {...props.getRootProps({refKey: 'innerRef'})}>
+                <SearchInput
+                    handleKeyDown={this.handleKeyDown}
+                    handleInputChange={this.handleInputChange}
+                    handleDelete={this.handleDelete}
+                    suggestions={suggestions}
+                    clickInput={this.clickInput}
+                    input={this.input}
+                    placeholder={placeholder}
+                    key={this.input}
```
Comment (jmorel) [inline src/app/business/common/components/searchBar/searchBar.js:262]: Same question as above, this looks like an addition.

File: src/app/business/routes/objective/components/detail/components/tabs/index.js [hunk 3]
```diff
@@ -3,19 +3,19 @@ import PropTypes from 'prop-types';
...
-} from '@substrafoundation/substra-ui';
 
-import Tab from '../../../../../../common/components/detail/components/tabs';
 import PermissionsTabPanelContent from '../../../../../../common/components/detail/components/permissionsTabPanelContent';
 import ForbiddenResource from '../../../../../../common/components/detail/components/forbiddenResource';
 import Description from '../../../../../../common/components/detail/components/description';
 import {spacingNormal} from '../../../../../../../../../assets/css/variables/spacing';
+import {
+    TabList,
+    Tab,
```
Comment (jmorel) [inline src/app/business/routes/objective/components/detail/components/tabs/index.js:13]: This was actually imported from `common/components/detail/components/tabs` beforehand. Could you check what the difference is between the two components and decide which one should be used (and if they are both needed?)

File: src/app/business/common/components/searchBar/searchBar.js (hunk 1 above)
Comment (maeldebon) [reply src/app/business/common/components/searchBar/searchBar.js:51]: I got errors that told me to add a key argument to this element.

File: src/app/business/common/components/searchBar/searchBar.js (hunk 2 above)
Comment (maeldebon) [reply src/app/business/common/components/searchBar/searchBar.js:262]: Yes indeed, I got the same errors for this one.

File: src/app/business/routes/objective/components/detail/components/tabs/index.js (hunk 3 above)
Comment (maeldebon) [reply src/app/business/routes/objective/components/detail/components/tabs/index.js:13]: It looks like the `common/components/detail/components/tabs` was a layer added to SubstraUI's Tab component that enabled us to change the tab's color following the front-end we were using. This file isn't used anymore, we could even remove it.

**Comment by jmorel** [comment]
When running the tests, I get the following warning:
```
 console.error node_modules/react-dom/cjs/react-dom.development.js:530
    Warning: Each child in a list should have a unique "key" prop. See https://fb.me/react-warning-keys for more information.
        in div (created by Context.Consumer)
        in Wrapper
        in CopyInput
```
This is triggered by the presence of a ref on Input in CopyInput, but the warning only happens in jest and not in the browser so I'm ignoring it for the moment.

**Pull request merged**